src/ArduinoIoTCloud.cpp

```diff
@@ -332,6 +404,11 @@ void ArduinoIoTCloudClass::connectionCheck()
         CloudSerial.begin(9600);
         CloudSerial.println("Hello from Cloud Serial!");
       }
+      #ifdef ARDUINO_ARCH_SAMD
+        unsigned long epoch = getTime();
```

Comment:
Please use **const correctness** as much as possible. Const everything and everywhere and the compiler will detect when variables are written too that are intended **read-only** by the programmer.
```suggestion
        unsigned long const epoch = getTime();
```
Also, for better expressiveness you could also give a name to the if condition:
```C++
bool const is_valid_epoch = epoch != 0;
if(is_valid_epoch) ...
```